feat(react-scheduler): add appointment resources support (#2498)
DevExpress/devextreme-reactive: packages/dx-react-scheduler/docs/reference/resources.md

# Resources Plugin Reference

A plugin that configure resources.

## Import

Use the following statement to import a plugin with embedded theme components:

```js
import { Resources } from '@devexpress/dx-react-scheduler-material-ui';
```

If you want to use custom components, you can import the themeless plugin:

```js
import { Resources } from '@devexpress/dx-react-scheduler';
```

## User Reference

### Dependencies

- [Appointments](appointments.md)

### Properties

Name | Type | Default | Description
-----|------|---------|------------

> data | Array&lt;[Resource](#resource)&gt; | The resources data set.

Reviewer: Description is in Default's place

## Interfaces

### Resource

Specifies a resource that available in scheduler.

Field | Type | Description
------|------|------------
id | number &#124; string | The resource kind ID.
fieldName | string | The appointment data field name that would binding to resource.
items | Array&lt;[ResourceItem](#resourceitem)&gt; | The array of resource items.
title? | string | The resource kind title that would displays inside UI components.
allowMultiple? | boolean | Indicates whether or not several resources of this kind can be assigned to an appointment.

### ResourceItem

Specifies the resource item object.

Field | Type | Description
------|------|------------
id | number &#124; string | The resource item ID.
text? | string | The resource item text that would displays inside UI components.
color? | string &#124; [Color](#color) | The resource item color that would used inside UI components.

### Palette

Specifies the resources palette.

Type: `Array<string | Color>`

### ValidResource

> 

Reviewer: Should we add a description here?

Field | Type | Description
------|------|------------
fieldName | string | The appointment data field name that would binding to resource.
text | string | The resource item text that would displays inside UI components.
title | string | The resource kind title that would displays inside UI components.
allowMultiple | boolean | Indicates whether or not several resources of this kind can be assigned to an appointment.
isMain | boolean | Specifies the main resource kind.
items | Array&lt;[ValidResourceItem](#validresourceitem)&gt; | The array of resource items.
color? | string &#124; [Color](#color) | The resource item color that would used inside UI components.

### ValidResourceItem

> 

Reviewer: Same here

# Resources Plugin Reference

A plugin that configure resources.

## Import

Use the following statement to import a plugin with embedded theme components:

```js
import { Resources } from '@devexpress/dx-react-scheduler-material-ui';
```

If you want to use custom components, you can import the themeless plugin:

```js
import { Resources } from '@devexpress/dx-react-scheduler';
```

## User Reference

### Dependencies

- [Appointments](appointments.md)

### Properties

Name | Type | Default | Description
-----|------|---------|------------
data | Array&lt;[Resource](#resource)&gt; | | The resources data set.
mainResourceName? | string | | Specifies whether appointments are colored like this resource kind.

> palette? | [Palette](#palette) | [red, pink, purple, deepPurple, indigo, blue, lightBlue, cyan, teal, green, lightGreen, lime, yellow, ember, orange, deepOrange] | The palette that would be used if resource item colors were not defined.

Author: You can set a color by a `string` variable.

> palette? | [Palette](#palette) | [red, pink, purple, deepPurple, indigo, blue, lightBlue, cyan, teal, green, lightGreen, lime, yellow, ember, orange, deepOrange] | The palette that would be used if resource item colors were not defined.

Author: No, these list contains Material-UI color objects

## Interfaces

### Resource

Configures a resource.

Field | Type | Description
------|------|------------
id | number &#124; string | The resource ID.
fieldName | string | A data field name used to assign appointments to this resource. This field should be present in appointment data objects.
instances | Array&lt;[ResourceInstance](#resourceinstance)&gt; | Resource instances.
title? | string | The resource title.
allowMultiple? | boolean | Indicates whether an appointment can be assigned to several instances of this resource.

### ResourceInstance

Configures a resource instance.

Field | Type | Description
------|------|------------
id | number &#124; string | The resource instance ID.
text? | string | The resource instance text.
color? | string &#124; [Color](#color) | The resource instance color.

### Palette

> Specifies a palette that provides colors for those resource instances whose color is not defined.

Reviewer 2: ```suggestion
Specifies a palette that provides colors for resource instances with undefined colors.
```
